src/index.ts

```diff
@@ -9,42 +9,45 @@ type TagList = Record<string, any>
 
 const log = debug('pyroscope')
 
+const cloudHostnameSuffix = 'pyroscope.cloud'
+
 export interface PyroscopeConfig {
-  server?: string
-  name: string
+  serverAddress: string
+  appName: string
   sourceMapPath?: string[]
-  autoStart: boolean
   sm?: any
   tags: TagList
+  authToken?: string
+  configured: boolean
 }
 
 const INTERVAL = 10000
 const SAMPLERATE = 100
 // Base sampling interval, constant for pyroscope
 const DEFAULT_SERVER =
-  process.env['PYROSCOPE_SERVER'] || 'http://localhost:4040'
+  process.env['PYROSCOPE_SERVER_ADDRESS'] || 'http://pyroscope:4040'
 
 const config: PyroscopeConfig = {
-  server: DEFAULT_SERVER,
-  autoStart: true,
-  name: 'nodejs',
+  serverAddress: DEFAULT_SERVER,
+  appName: process.env['PYROSCOPE_APPLICATION_NAME'] || 'nodejs',
   sm: undefined,
   tags: {},
+  authToken: process.env['PYROSCOPE_AUTH_TOKEN'],
+  configured: false,
 }
 
-
 export function init(
   c: Partial<PyroscopeConfig> = {
-    server: DEFAULT_SERVER,
-    autoStart: true,
-    name: 'nodejs',
+    serverAddress: DEFAULT_SERVER,
+    appName: 'nodejs',
     tags: {},
   }
 ): void {
   if (c) {
-    config.server = c.server || DEFAULT_SERVER
+    config.serverAddress = c.serverAddress || DEFAULT_SERVER
     config.sourceMapPath = c.sourceMapPath
-    config.name = c.name || 'nodejs'
+    config.appName = c.appName || 'nodejs'
```

Comment:
Based on the go implementation, `applicationName` is required https://github.com/pyroscope-io/client/blob/563bf89a859604a1cca9bc01503e8b4de50b6f05/internal/flameql/error.go#L15

I guess we should do the same?

Reply:
Made app name mandatory